src/storage/conversion/BaseTypedRepresentationConverter.ts

```diff
@@ -0,0 +1,76 @@
+import type { ValuePreferences } from '../../http/representation/RepresentationPreferences';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import { getConversionTarget, getTypeWeight, preferencesToString } from './ConversionUtil';
+import type { RepresentationConverterArgs } from './RepresentationConverter';
+import { TypedRepresentationConverter } from './TypedRepresentationConverter';
+
+type PromiseOrValue<T> = T | Promise<T>;
+type ValuePreferencesArg =
+  PromiseOrValue<string> |
+  PromiseOrValue<string[]> |
+  PromiseOrValue<ValuePreferences>;
+
+async function toValuePreferences(arg: ValuePreferencesArg): Promise<ValuePreferences> {
+  const resolved = await arg;
+  if (typeof resolved === 'string') {
+    return { [resolved]: 1 };
+  }
+  if (Array.isArray(resolved)) {
+    return Object.fromEntries(resolved.map((type): [string, number] => [ type, 1 ]));
+  }
+  return resolved;
+}
+
+/**
+ * A {@link TypedRepresentationConverter} that can convert from all its input types to all its output types.
+ */
+export abstract class BaseTypedRepresentationConverter extends TypedRepresentationConverter {
```

Comment:
I'd need a bit more explanation of what to use this base class for.

Reply:
Would it make sense to make this the `TypedRepresentationConverter` implementation?
I feel like I am missing something.

Reply:
I just wanted to differentiate the interface from the base implementation that is used by several converters. This base class is just 1 specific way to implement the typed converter.

Reply:
^ perhaps add this to the comment?